### ISSUE TITLE
Rule Request: Prevent moot unwrapped variable in guard/if-let

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#5025]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

I countered some cases in my codebase where I unwrap a variable in a guard/if-let, but then don't use it directly.

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

It prevents moot code.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

This:

```swift
guard
	let asset = asset, // asset is not used
	let date = asset.creationDate
else {
	return
}

search(withQuery: date)
```

Could simply be:

```swift
guard
	let date = asset?.creationDate
else {
	return
}

search(withQuery: date)
```

> 4. Should the rule be configurable, if so what parameters should be configurable?

No

> 5. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Having moot code is not harmful, so I guess it could be opt-in.

**Comment by roxanamihaela199** [comment]
@SimplyDanny

**Comment by roxanamihaela199** [comment]
![Screenshot_2023-06-01-05-43-31-286_com video fun app](https://github.com/realm/SwiftLint/assets/115851657/31cd3ee5-7add-4050-8973-793b14a8c8ed)

**Comment by SimplyDanny** [comment]
Would it only trigger on `let x = x` or on `let x` and `let x = <complicated expression>` as well given that `x` isn't used anywhere outside of the `guard` statement later on?

**Comment by sindresorhus** [comment]
It would trigger on all of these.
